tests/e2e/test_methods/rds_s3.py

```diff
@@ -0,0 +1,363 @@
+
+    mysql_client = get_mysql_client(
+        user=db_username,
+        password=db_password,
+        host=rds_endpoint,
+        database="mlpipeline",
+    )
+
+    resp = mysql_utils.query(
+        mysql_client, f"select * from experiments where Name='{name}'"
+    )
+    assert len(resp) == 1
+    assert resp[0]["Name"] == experiment.name
+    assert resp[0]["Description"] == experiment.description
+    assert resp[0]["Namespace"] == experiment.resource_references[0].key.id
+
+    resp = kfp_client.get_experiment(
+        experiment_id=experiment.id, namespace=user_namespace
+    )
+
+    assert name == resp.name
+    assert description == resp.description
+    assert user_namespace == resp.resource_references[0].key.id
+
+    kfp_client.delete_experiment(experiment.id)
+
+    resp = mysql_utils.query(
+        mysql_client, f"select * from experiments where Name='{name}'"
+    )
+    assert len(resp) == 0
+
+    try:
+        kfp_client.get_experiment(experiment_id=experiment.id, namespace=user_namespace)
+        raise AssertionError("Expected KFPApiException Not Found")
+    except KFPApiException as e:
+        assert "Not Found" == e.reason
+
+    mysql_client.close()
+
+
+def test_run_pipeline(
+    kfp_client,
+    s3_bucket_name,
+    db_username,
+    db_password,
+    rds_endpoint,
+    region,
+    user_namespace=DEFAULT_USER_NAMESPACE,
+    pipeline_name=PIPELINE_XG_BOOST,
+):
+    s3_client = get_s3_client(region)
+
+    experiment_name = rand_name("experiment-")
+    experiment_description = rand_name("description-")
+    experiment = kfp_client.create_experiment(
+        experiment_name,
+        description=experiment_description,
+        namespace=user_namespace,
+    )
+
+    pipeline_id = kfp_client.get_pipeline_id(pipeline_name)
+    job_name = rand_name("run-")
+
+    run = kfp_client.run_pipeline(
+        experiment.id, job_name=job_name, pipeline_id=pipeline_id
+    )
+
+    assert run.name == job_name
+    assert run.pipeline_spec.pipeline_id == pipeline_id
+
+    resp = wait_for_run_succeeded(kfp_client, run, job_name, pipeline_id)
+
+    workflow_manifest_json = resp.pipeline_runtime.workflow_manifest
+    workflow_manifest = json.loads(workflow_manifest_json)
+
+    s3_artifact_keys = []
+    for _, node in workflow_manifest["status"]["nodes"].items():
+        if "outputs" not in node:
+            continue
+        if "artifacts" not in node["outputs"]:
+            continue
+
+        for artifact in node["outputs"]["artifacts"]:
+            if "s3" not in artifact:
+                continue
+            s3_artifact_keys.append(artifact["s3"]["key"])
+
+    bucket_objects = s3_client.list_objects_v2(Bucket=s3_bucket_name)
+    content_keys = {content["Key"] for content in bucket_objects["Contents"]}
+
+    assert f"pipelines/{pipeline_id}" in content_keys
+    for key in s3_artifact_keys:
+        assert key in content_keys
+
+    mysql_client = get_mysql_client(
```

Comment:
should also add a assert for few more tables later on, atleast artifact based on #327

Reply:
added todo